02_bubble/alfredxiao/main.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func makeCopy(ns []int) []int {
+	return append([]int{}, ns...)
+}
+
+func bubble(s []int) []int {
+	copy := makeCopy(s)
+	n := len(copy)
+	for i := n - 1; i > 0; i-- {
+		for j := 0; j < i; j++ {
+			maybeSwap(copy, j, j+1)
+		}
+	}
+	return copy
+}
+
+func maybeSwap(ns []int, i, j int) bool {
```

Comment:
name is ugly. I'd prefer sth like `swapIfLower`

Reply:
I have found some people are not comfortable with names in the `maybeVerb` form. But this is actually common. If you search in JDK source code, you can find many occurrences of maybe* there. Although I agree swapIfLower is a good candidate, naming is always a very difficult and subjective thing. You can actually find mongodb uses maybeSwap in https://github.com/mongodb/mongo/search?q=maybeSwap&unscoped_q=maybeSwap

Reply:
Clean code says function name should be descriptive and blah blah blah (holly war, let's skip this :) ).
Don't like this but it's not a blocker in my mind.